features/include/pcl/features/impl/multiscale_feature_persistence.hpp

```diff
@@ -166,12 +166,11 @@ pcl::MultiscaleFeaturePersistence<PointSource, PointFeature>::extractUniqueFeatu
   unique_features_indices_.reserve (scale_values_.size ());
   unique_features_table_.reserve (scale_values_.size ());
 
+  std::vector<float> diff_vector;
```

Comment:
Some heuristic based reserve for `diff_vector` as well?

Reply:
I will move this to it's previous position, it will still work! Thanks

Reply:
Maybe better to use clear first followed by a reserve, what do you think? Just like the previous way, but the order is different. Because I wasnt sure whether clear actually deletes the allocated capacity reserved by the reserve function, and i couldn't find an exact answer. If it deletes then it doesn't make sense to do reserve and clear it

Reply:
Added a comment on the site of the reserve. Didn't see this till it was too late